File: dener8/programmers_124나라의숫자.py
'''
1, 2, 4로만 구성
ans += "1" or "2" or "4"
이 후에 뒤집에서 return
'''

def solution(n):
    answer = ''

    while n:  # 조정 필요
        if n % 3 == 0:
            answer += '4'
        elif n % 3 == 1:
            answer += '1'
        elif n % 3 == 2:
            answer += '2'
        n = (n - 1) // 3

    return answer[::-1]

File: myunghyunnero/programmers_124나라의숫자.cpp
#include <string>
#include <vector>

using namespace std;

string solution(int n) {
    string answer = "";
    int remain;
    
    while (n!=0) {
        remain = n % 3;
        n /= 3;

        if (remain == 0) { //나누어 떨어질 경우 몫을 1빼준다.
            answer = "4" + answer;
            n--;
        }
        else if (remain == 1) {
            answer = "1" + answer;
        }
        else if (remain == 2) {
            answer = "2" + answer;
        }
    }
    return answer;
}